sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileServiceSasSignatureValues.java

```diff
@@ -216,65 +233,43 @@ public FileServiceSasSignatureValues setSasIpRange(SasIpRange sasIpRange) {
      *
-     * @param resource Allowed resources string to set
-     * @return the updated FileServiceSasSignatureValues object
+     * @return The name of the share being made accessible.
      */
-    public FileServiceSasSignatureValues setResource(String resource) {
-        this.resource = resource;
-        return this;
+    public String getShareName() {
+        return shareName;
     }
 
     /**
-     * @return the canonical name of the object the SAS user may access.
-     */
-    public String getCanonicalName() {
-        return canonicalName;
-    }
-
-    /**
-     * Sets the canonical name of the object the SAS user may access.
+     * Sets the name of the share being made accessible.
      *
-     * @param canonicalName Canonical name of the object the SAS grants access
+     * @param shareName The name of the share being made accessible.
      * @return the updated FileServiceSasSignatureValues object
      */
-    public FileServiceSasSignatureValues setCanonicalName(String canonicalName) {
-        this.canonicalName = canonicalName;
+    public FileServiceSasSignatureValues setShareName(String shareName) {
+        this.shareName = shareName;
         return this;
     }
 
     /**
-     * Sets the canonical name of the object the SAS user may access. Constructs a canonical name of
-     * "/file/{accountName}/{shareName}/{filePath}".
+     * Gets the path of the file or directory being made accessible. {@code null} or an empty string for a share SAS.
      *
-     * @param shareName Name of the share
-     * @param filePath Name of the file
-     * @param accountName Name of the account that contains the object
-     * @return the updated FileServiceSasSignatureValues object
+     * @return The path of the file or directory being made accessible. {@code null} or an empty string for a share SAS.
      */
-    public FileServiceSasSignatureValues setCanonicalName(String shareName, String filePath, String accountName) {
-        this.canonicalName = String.format("/file/%s/%s/%s", accountName, shareName, filePath);
-        return this;
+    public String getFilePath() {
+        return filePath;
     }
 
     /**
-     * Sets the canonical name of the object the SAS user may access. Constructs a canonical name of
-     * "/file/{accountName}/{shareName}".
+     * Sets the path of the file or directory being made accessible. Pass in {@code null} or an empty string for a share
+     * SAS.
      *
-     * @param shareName Name of the share
-     * @param accountName Name of the account that contains the object
+     * @param filePath The name of the share being made accessible.
      * @return the updated FileServiceSasSignatureValues object
      */
-    public FileServiceSasSignatureValues setCanonicalName(String shareName, String accountName) {
-        this.canonicalName = String.format("/file/%s/%s", accountName, shareName);
+    public FileServiceSasSignatureValues setFilePath(String filePath) {
```

Comment:
I updated it to filePath because the other languages, [comparison with languages table](https://github.com/Azure/azure-sdk-for-java/issues/5745#issuecomment-541433173), had the same notation. I think this public change would have to be discussed with others.